apps/web/server/routers/donation.ts

```diff
@@ -0,0 +1,44 @@
+import { protectedProcedure, router } from "@server/trpc";
+import { serverEnv } from "@env/server";
+import {
+  AddressFormat,
+  MessageFormat,
+  AmountFormat,
+  NicknameFormat,
+} from "@server/input-formats";
+import { z } from "zod";
+import { TRPCError } from "@trpc/server";
+import { DEFAULT_TEST_DONATION } from "@donum/shared/constants";
+
+export const donationRouter = router({
+  sendTestDonation: protectedProcedure
+    .input(
+      z
+        .object({
+          from: NicknameFormat.optional(),
+          to: AddressFormat,
+          amount: AmountFormat,
+          message: MessageFormat.optional(),
+        })
+        .optional()
+    )
+    .mutation(async ({ ctx, input }) => {
+      const address = ctx.session.user.name!; // protectedProcedure always returns existing user
+      const donation = input || { ...DEFAULT_TEST_DONATION, to: address };
```

Comment:
здесь лучше сначала создать дефолтный донат, а потом заполнить его полями, которые есть. Вроде в js это можно сделать через
```suggestion
      const donation = { to: address, ...DEFAULT_TEST_DONATION,  ...input };
```